### PR TITLE
Upstream documentation update up-to 2022-02-16

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

**Comment by hamishwillee** [comment]
@amilcarlucas Thanks! Can you provide the generated files - just to make it easier for me to get my head around?

File: doc/mavlink_gitbook.py [hunk 1]
```diff
@@ -40,17 +40,48 @@
...
 
 > **Note** Vendor forks of MAVLink may contain dialect messages that are not yet merged, and hence will not appear in this documentation.
 
-The dialect files are stored alongside in separate XML files in [mavlink/message definitions](https://github.com/mavlink/mavlink/blob/master/message_definitions/).
+Human-readable forms of all the the core dialects are linked below:
+"""
+
+index_text_trailer="""## External Dialects
+
+MAVLink provides the [/external/dialects](https://github.com/mavlink/mavlink/tree/master/external/dialects) folder for dialects from projects that are not maintained by core MAVLink stakeholders or part of the MAVLink standard.
```
Comment (hamishwillee) [inline doc/mavlink_gitbook.py:76]: We don't provide this anymore. It was removed.

Comment (amilcarlucas) [reply doc/mavlink_gitbook.py:76]: I got this patch from upstream. It needs to be removed from upstream then.

**Comment by amilcarlucas** [comment]
AFAIK ardupilot does not generate, nor publishes any documentation files

**Comment by hamishwillee** [comment]
Arrggg. I'm a bit dim. I was assuming this was a change to mavlink/mavlink. I think I will need to fix up that doc on external dialects. Will notify myself to look at it tomorrow.

File: doc/mavlink_gitbook.py (hunk 1 above)
Comment (hamishwillee) [reply doc/mavlink_gitbook.py:76]: Done. Fixed by https://github.com/mavlink/mavlink/pull/1802

**Comment by amilcarlucas** [comment]
I have added the upstream fixes here

**Review by peterbarker — APPROVED** [review]

**Review by tridge — APPROVED** [review]

**Pull request merged**